.github/workflows/my-ci-pipeline.yml

```
@@ -16,3 +16,26 @@ jobs:
     - uses: actions/checkout@v2
     - run: docker build --target test --tag todo-app:test .
     - run: docker run todo-app:test
+
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+    steps:
+      - run: echo Publishing!
+      -
+        name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}      
+
+      -
+        name: Build and Push Production to Docker Hub
+        uses: actions/checkout@v2
```

```suggestion
        uses: actions/checkout@v4
```
Our advice was running a couple of versions behind earlier, which is the cause of the deprecation warnings that show up on your pipelines 🙂